### PR TITLE
Fixes #21963: Make rudderc technique able to be used by the technique editor

### PR DESCRIPTION
**Pull request by amousset**

### CODE REVIEW

File: policies/rudderc/src/backends/metadata.rs [hunk 1]
```diff
@@ -4,87 +4,298 @@
...
+impl Section {
+    fn from(resources: Vec<ResourceKind>) -> Vec<Self> {
+        // recursive function to handle blocks. Block recursion levels shouldn't be a
+        // stack overflow threat.
+        resources
+            .into_iter()
+            .map(|r| match r {
+                ResourceKind::Block(b) => Section {
+                    name: b.name,
+                    id: b.id,
```
Comment (Fdall) [inline policies/rudderc/src/backends/metadata.rs:178]: Why is this the only id that is not cloned?

Comment (amousset) [reply policies/rudderc/src/backends/metadata.rs:178]: Because it is moved (as we do not reuse it after)

**Review by Fdall — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**